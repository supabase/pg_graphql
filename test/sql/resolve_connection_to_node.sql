begin;

    create table account(
        id serial primary key,
        email varchar(255) not null
    );


    insert into public.account(email)
    values
        ('aardvark@x.com'),
        ('bat@x.com'),
        ('cat@x.com'),
        ('dog@x.com'),
        ('elephant@x.com');


    create table blog(
        id serial primary key,
        owner_id integer not null references account(id)
    );


    insert into blog(owner_id)
    values
        ((select id from account where email ilike 'a%')),
        ((select id from account where email ilike 'a%')),
        ((select id from account where email ilike 'a%')),
        ((select id from account where email ilike 'b%'));


    select jsonb_pretty(
        gql.resolve($$
        {
          allBlogs {
            edges {
              node {
                ownerId
                owner {
                  id
                }
              }
            }
          }
        }
        $$)
    );

rollback;
