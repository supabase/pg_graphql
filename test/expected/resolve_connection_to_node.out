begin;
    create table account(
        id serial primary key,
        email varchar(255) not null
    );
    insert into public.account(email)
    values
        ('aardvark@x.com'),
        ('bat@x.com'),
        ('cat@x.com'),
        ('dog@x.com'),
        ('elephant@x.com');
    create table blog(
        id serial primary key,
        owner_id integer not null references account(id)
    );
    insert into blog(owner_id)
    values
        ((select id from account where email ilike 'a%')),
        ((select id from account where email ilike 'a%')),
        ((select id from account where email ilike 'a%')),
        ((select id from account where email ilike 'b%'));
    select jsonb_pretty(
        gql.resolve($$
        {
          allBlogs {
            edges {
              node {
                ownerId
                owner {
                  id
                }
              }
            }
          }
        }
        $$)
    );
             jsonb_pretty             
--------------------------------------
 {                                   +
     "data": {                       +
         "allBlogs": {               +
             "edges": [              +
                 {                   +
                     "node": {       +
                         "owner": {  +
                             "id": 1 +
                         },          +
                         "ownerId": 1+
                     }               +
                 },                  +
                 {                   +
                     "node": {       +
                         "owner": {  +
                             "id": 1 +
                         },          +
                         "ownerId": 1+
                     }               +
                 },                  +
                 {                   +
                     "node": {       +
                         "owner": {  +
                             "id": 1 +
                         },          +
                         "ownerId": 1+
                     }               +
                 },                  +
                 {                   +
                     "node": {       +
                         "owner": {  +
                             "id": 2 +
                         },          +
                         "ownerId": 2+
                     }               +
                 }                   +
             ]                       +
         }                           +
     },                              +
     "errors": [                     +
     ]                               +
 }
(1 row)

rollback;
